relay/sources/relayd/src/shared_files.rs

```diff
@@ -235,13 +242,40 @@ pub fn file_writer(buf: &mut warp::body::FullBody, path: &str) {
     fs::write(format!("shared-files/{}", path), myvec).expect("Unable to write file");
 }
 
-pub fn metadata_writer(metadata_string: String, peek: &str) {
+pub fn put_handler(
+    metadata_string: String,
+    peek: &str,
+    ttl: String,
+    job_config: Arc<JobConfig>,
+    mut buf: warp::body::FullBody,
+) -> hyper::StatusCode {
+    let timestamp = parse_ttl(ttl);
+
+    if timestamp.is_err() {
+        return StatusCode::from_u16(500).unwrap();
+    }
+
     let myvec: Vec<String> = peek.split('/').map(|s| s.to_string()).collect();
     let (target_uuid, source_uuid, _file_id) = (&myvec[0], &myvec[1], &myvec[2]);
+
+    if !job_config
+        .nodes
+        .read()
+        .expect("Cannot read nodes list")
+        .is_subnode(source_uuid)
+    {
+        return StatusCode::from_u16(404).unwrap();
+    }
+
     let _ = fs::create_dir_all(format!("shared-files/{}/{}/", target_uuid, source_uuid)); // on cree les folders s'ils existent pas
                                                                                           //fs::create_dir_all(format!("/var/rudder/configuration-repository/shared-files/{}/{}/", target_uuid, source_uuid)); // real path
-    fs::write(format!("shared-files/{}.metadata", peek), metadata_string)
-        .expect("Unable to write file");
+    fs::write(
+        format!("shared-files/{}.metadata", peek),
+        format!("{}expires={}\n", metadata_string, timestamp.unwrap()),
```

Comment:
we shouldn't unwrap here, we need to properly manage the error

Reply:
I just saw it is actually correct as the error cases returns early. Could you change the definition of timestamp to something like ```rust
let timestamp = match parse_ttl(..) {
    Ok(ttl) => ttls,
    Err(_) => return StatusCode::from_u16(500).unwrap(),
}```

to make it clearer?
}